paddle/fluid/framework/ir/graph_pattern_detector.cc

```diff
@@ -983,9 +983,12 @@ PDNode *patterns::QuantTranspose2::operator()() {
   auto *quant_in = pattern->NewNode(quant_in_repr())
                        ->AsInput()
                        ->assert_is_op_input("quantize", "Input");
-  auto *quant_op = pattern->NewNode(quant_op_repr())->assert_is_op("quantize");
+  auto *quant_op = pattern->NewNode(quant_op_repr())
+                       ->assert_is_op("quantize")
+                       ->AsIntermediate();
```

Comment:
Same here, ops don't seem to have roles. I think they only concern links. However, I don't know what all the code is doing in this file.
![image](https://user-images.githubusercontent.com/17185347/216379795-b2b0ccc8-989b-4d04-8e66-153008a88f32.png)

Reply:
If you found out that we should add AsIntermediate to removed nodes, it would be great and we could fix all patterns.

Reply:
@sfraczek  From what I've found in code AsIntermediate is used only during subgraph validation - adding AsIntermediate prevents node from being used in multiple overlapping patterns or linked with external nodes.

Reply:
Is it only for vars though or ops too?